packages/dev/core/src/Materials/Textures/renderTargetTexture.ts

```diff
@@ -1300,7 +1299,9 @@ export class RenderTargetTexture extends Texture implements IRenderTargetTexture
 
             // Dump ?
             if (dumpForDebug) {
-                DumpFramebuffer(this.getRenderWidth(), this.getRenderHeight(), engine);
+                import("../../Misc/dumpTools").then((dumpTools) => {
+                    dumpTools.DumpFramebuffer(this.getRenderWidth(), this.getRenderHeight(), engine);

```

Comment:
It won't work, because we are in a promise when this code executes, so in a js job outside the main js flow. At that time, the currently bound framebuffer is likely different from the framebuffer we wanted to dump. Also, getRenderWidth/Height returns the width/height of the currently bound framebuffer. So, something like this *may* work, but you will have to test to be sure:

```typescript
                const width = this.getRenderWidth(),
                    height = this.getRenderHeight();
                import("../../Misc/dumpTools").then((dumpTools) => {
                    const currentRenderTarget = engine._currentRenderTarget;
                    if (currentRenderTarget !== this.renderTarget) {
                        engine.bindFramebuffer(this.renderTarget!);
                    }
                    dumpTools.DumpFramebuffer(width, height, engine);
                    if (currentRenderTarget !== this.renderTarget) {
                        if (currentRenderTarget) {
                            engine.bindFramebuffer(currentRenderTarget);
                        } else {
                            engine.restoreDefaultFramebuffer();
                        }
                    }
                });
```

Reply:
You are totally right. A different solution would be to async-load the module before and just use the reference like before. This way we keep the same behavior and just avoid a sync import.